File: event-service/src/main/java/com/github/manosbatsis/services/event/repository/UserEventRepository.java
package com.github.manosbatsis.services.event.repository;

import com.github.manosbatsis.services.event.model.UserEvent;
import com.github.manosbatsis.services.event.model.UserEventKey;

import org.springframework.data.cassandra.repository.CassandraRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface UserEventRepository extends CassandraRepository<UserEvent, UserEventKey> {

    List<UserEvent> findByKeyUserId(Long id);
}

File: event-service/src/main/kotlin/com/github/manosbatsis/services/event/repository/UserEventRepository.kt
package com.github.manosbatsis.services.event.repository

import com.github.manosbatsis.services.event.model.UserEvent
import com.github.manosbatsis.services.event.model.UserEventKey
import org.springframework.data.cassandra.repository.CassandraRepository
import org.springframework.stereotype.Repository

@Repository
interface UserEventRepository : CassandraRepository<UserEvent, UserEventKey> {
    fun findByKeyUserId(id: Long): List<UserEvent>
}
